api/src/main/java/ru/urururu/sanity/cpp/Demangler.java

```diff
@@ -0,0 +1,50 @@
+package ru.urururu.sanity.cpp;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+public class Demangler {
+    public static String demangle(String mangled) {
+        String prefix = "_ZN";
+        if (!mangled.startsWith(prefix)) {
+            return mangled;
+        }
+
+        char[] chars = mangled.toCharArray();
+
+        int pos = prefix.length();
+
+        List<String> parts = new ArrayList<>();
+
+        while (pos < chars.length) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS135)